BEST_PRACTICES.md

```diff
@@ -0,0 +1,18 @@
+Follow these best practices while developing any plugin:
```

Comment:
I think this should just be rolled into STANDARDS, cause those are all "best practices" (e.g. using a link checker). I would call this section "Development Best Practices".

Reply:
> Thanks! Is this for OpenSearch Plugins, OpenSearch Dashboard Plugins or both? I think we want sections specifically for both. Also let's merge this into STANDARDS?
> 
> For every section ideally we should have "good" vs. "bad", or links to PRs that show how something is done well.

This is true for any plugin development which takes place. Ideally the design principles mentioned here are for the backend app development but these best practices can be followed for dashboards as well.

I will circle back once I have PRs demonstrating these design principles and add them here.